src/httpattack/httpattack.go

```diff
@@ -0,0 +1,273 @@
+package httpattack
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"log"
+	"math/rand"
+	"net"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/corpix/uarand"
+	"github.com/google/uuid"
+	"github.com/valyala/fasthttp"
+	"github.com/valyala/fasthttp/fasthttpproxy"
+
+	"github.com/Arriven/db1000n/src/metrics"
+	"github.com/Arriven/db1000n/src/utils"
+	"github.com/Arriven/db1000n/src/utils/templates"
+)
+
+func NewHttpAttackClient(clientCfg map[string]interface{}, proxyURL string, debug bool) (client *HttpAttackClient) {
```

Comment:
I do like the idea of the change but it's better to expose ClientConfig interface to the world and let the job handle parsing (see how it's done in packetgen)